Makefile

```diff
@@ -347,19 +347,23 @@ ifeq ($(UNAME_S),Darwin)
     HOST_OS:=darwin
 endif
 
+HOST_BIN_PLATFORM := bin/$(HOST_OS)_$(GOARCH)
+
 .PHONY: build-prep
 build-prep: deps
 	mkdir -p bin/{darwin_amd64,darwin_arm64,linux_amd64,linux_arm64,linux_ppc64le,linux_s390x,windows_amd64}
 
 .PHONY: cli-build
 cli-build: build-prep
+ifndef CI
+	RACE=0 CGO_ENABLED=0 GOOS=$(HOST_OS) GOARCH=$(GOARCH) $(GOBUILD) ./roxctl
```

Comment:
Yeah so for Darwin and arm64, will need to create both linux/arm64 and darwin/arm64 so that local roxctl will work and then the image will work as well because that is linux/arm64

Reply:
Actually, nvm, it seems to work correctly due to the later build :)

Reply:
Somehow, there's `main-build-nodeps` target that takes care of linux/$(GOARCH) version when `CI` isn't set.

Reply:
So, since we are cleaning up here, there's maybe a more fundamental issue we should fix:
There are two (three) use cases for building `roxctl`:
1. Building `roxctl` for the host platform, for use in scripts etc.
2. Building `roxctl` for all platforms for embedding in the image
3. (Building `roxctl` for the target platform for a quicker CI build)

2 and 3 are alternatives and 1 might overlap with 2/3.

However, the current set of targets doesn't follow that structure. Instead, there's a target to accomplish 1+2 in CI or just 1 in non-CI, and 3 is (if desired) accomplished via `main-build-nodeps`. Also, 1 is accomplished in conjunction with the target `cli`, of which `cli-build` (that might also accomplish 2 in CI, but not in non-CI) is a dependency.

I'd propose to restructure the target such that `cli-build` addresses 2/3 (i.e., in the `ifndef CI` case, build `roxctl` not for the _host_ but for the _target_ platform), make `cli-build` a direct dependency of `main-build-nodeps` (because it is responsible for producing the Go binaries that go into the main image; if you take offence at the fact that `-nodeps` has a dependency, just build it as a sub-target via `$(MAKE) cli-build`), remove the conditional build of `roxctl` from `main-build-nodeps`, and rephrase `cli` in terms of `go install` and make it build `roxctl` for the _host_ platform only.

IMHO this would be much clearer. This means we might build `roxctl` twice for the host platform (once for embedding in the image if the host platform matches the target platform or we're in CI), but at least the binaries in the `bin/{platform}` directories will never be in conflict, because we circumvent this for the (1) case via `go install`. I don't think there's an easy way to avoid this duplication and it seems tolerable; the only way around it seems to be to introduce a dedicated target for _each_ platform.

Reply:
@misberner Could you please take a look at recent commits and tell if I'm going in the right direction?
There may be CI failures, test failures, absent reusable values and other things at the moment, but the most important for me at this point is to validate if I got your idea right.